File: contests/slpr1/p3/p3_solution.cpp
#include <iostream>
#include <vector>

typedef unsigned long long ull;

int main() {
    std::ios_base::sync_with_stdio(false);
    std::cin.tie(NULL);
    
    int n, q;
    std::cin >> n >> q;
    
    int d_psa[1000000];
    int p_psa[1000000];

    for (int i = 1; i <= n; ++i) {
        int x;
        std::cin >> x;
        d_psa[i]=d_psa[i-1]+x;
    }
    
    for (int i = 1; i <= n; ++i) {
        int x;
        std::cin >> x;
        p_psa[i]=p_psa[i-1]+x;
    }

    for (int i = 0; i < q; ++i) {
        int l, r;
        std::cin >> l >> r;

        ull ds = d_psa[r]-d_psa[l-1];
        ull ps = p_psa[r]-p_psa[l-1];

        if (ds==ps) {
            std::cout << "WOOHOO!\n";
        } else {
            std::cout << "BOOHOO!\n";
        }
    }
}

File: contests/slpr1/p3/p3_solution.py
import sys
raw_input = sys.stdin.readline
n,q = map(int, raw_input().split())
d_values=[int(i) for i in raw_input().split()]
p_values=[int(i) for i in raw_input().split()]

d_psa=[0]
p_psa=[0]

for i in range(1,n+1):
    d_psa.append(d_psa[i-1]+d_values[i-1])
    p_psa.append(p_psa[i-1]+p_values[i-1])
    
for i in range(q):
    l,r = map(int,raw_input().split())
    ds = d_psa[r]-d_psa[l-1]
    ps = p_psa[r]- p_psa[l-1]
    
    if ds == ps:
        print 'WOOHOO!'
    else:
        print 'BOOHOO!'
